File: solo-leveling-ui-service/src/main/java/com/sleepkqq/sololeveling/ui/broadcast/Topic.java
package com.sleepkqq.sololeveling.ui.broadcast;

import com.sleepkqq.sololeveling.avro.notification.Notification;
import com.vaadin.flow.shared.Registration;

public interface Topic<T> {

  Registration subscribe(long userId, T listener);

  void publish(long userId, Notification notification);
}

File: solo-leveling-ui-service/src/main/kotlin/com.sleepkqq.sololeveling/ui/broadcast/Topic.kt
package com.sleepkqq.sololeveling.ui.broadcast

import com.sleepkqq.sololeveling.avro.notification.Notification
import com.vaadin.flow.shared.Registration

interface Topic<T> {

	fun subscribe(userId: Long, listener: T): Registration

	fun publish(userId: Long, notification: Notification)
}
